File: problems/2522_partition_string_into_substrings_with_values_at_most_k/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    int minimumPartition(string s, int k) {
        int parts = 1; long long cur = 0;
        for (char c : s) {
            int d = c - '0'; if (d > k) return -1;
            cur = cur * 10 + d;
            if (cur > k) { parts++; cur = d; }
        }
        return parts;
    }
};

File: problems/2522_partition_string_into_substrings_with_values_at_most_k/solution.py
class Solution:
    def minimumPartition(self, s: str, k: int) -> int:
        parts, cur = 1, 0
        for c in s:
            d = int(c)
            if d > k: return -1
            cur = cur * 10 + d
            if cur > k:
                parts += 1
                cur = d
        return parts
